napari/utils/events/dataclass.py

```diff
@@ -225,8 +252,13 @@ def evented_post_init(self: T, *initvars) -> None:
             self.events.add(**e_fields)
         else:
             self.events = EmitterGroup(
-                source=self, auto_connect=False, **e_fields
+                source=self,
+                auto_connect=False,
+                deprecated=getattr(cls, "deprecated_events", None),
+                **e_fields,
             )
+        if not hasattr(self, "_hidden_event"):
+            self._hidden_event = Event(self)
```

Comment:
I'm a little nervous about this `self._hidden_event`. I assume it was necessary for some reason, but I don't immediately see why

Reply:
Try to debugging macOS build fail.